gax_test.go

```diff
@@ -1,299 +1,53 @@
-	}
-	for i, testCase := range invokeTestCases {
-		callCount := 0
-		_, err := Invoke(context.Background(), nil, func(ctx context.Context, req interface{}) (interface{}, error) {
-			callCount++
-			return nil, testCase.apiCall(callCount)
-		}, testOptions...)
-		if callCount != testCase.expectedCalls {
-			t.Errorf("[%d]: Invoke doesn't call the function properly: %d", i, callCount)
-		}
-		if grpc.Code(err) != testCase.expectedErrorCode {
-			t.Errorf("[%d]: Invoke returns error %v but expected error code is %v", i, err, testCase.expectedErrorCode)
-		}
-	}
-}
-
-func waitForContext(t *testing.T, ctx context.Context) error {
-	if _, ok := ctx.Deadline(); !ok {
-		t.Errorf("child context should have deadline")
-		return grpc.Errorf(codes.InvalidArgument, "")
-	}
-	<-ctx.Done()
-	return grpc.Errorf(codes.DeadlineExceeded, "")
-}
-
-func TestInvokeTimeout(t *testing.T) {
-	testOptions := getTestOptions()
-	opt := defaultCallOpt()
-	testCases := []struct {
-		parentTimeout         float64
-		expectedCalls         int
-		parentFailureExpected bool
-		expectedErrorCode     codes.Code
-	}{
-		{10, opt.maxAttempts, false, codes.DeadlineExceeded},
-		{3, 2, true, codes.OK},
-		{1.5, 1, true, codes.OK},
-		{0.25, 1, true, codes.OK},
-	}
-
-	callCount := 0
-	waitFunc := func(ctx context.Context, req interface{}) (interface{}, error) {
-		callCount++
-		return nil, waitForContext(t, ctx)
-	}
-
-	for i, testCase := range testCases {
-		callCount = 0
-		parent_context, _ := context.WithTimeout(context.Background(), time.Duration(testCase.parentTimeout*float64(timeUnit)))
-		_, err := Invoke(parent_context, nil, waitFunc, testOptions...)
-		if callCount != testCase.expectedCalls {
-			t.Errorf("[%d]: Invoke() invokes the function %d times (expected: %d)", i, callCount, testCase.expectedCalls)
-		}
-		if testCase.parentFailureExpected {
-			if err == nil || err != parent_context.Err() {
-				t.Errorf("[%d]: The overall context for Invoke() failure is expected, but got %v (ovreall context error is %v", i, err, parent_context.Err())
-			}
-		} else {
-			if grpc.Code(err) != testCase.expectedErrorCode {
-				t.Errorf("[%d]: Error code %v is expected, but got an error %v", testCase.expectedErrorCode, err)
-			}
-		}
-	}
-}
-
-func TestLongInterval(t *testing.T) {
-	testOptions := getTestOptions()
-	testOptions = append(testOptions, WithIntervalInfo(time.Minute, 2*time.Minute, 1.0))
-	waitFunc := func(ctx context.Context, req interface{}) (interface{}, error) {
-		return nil, waitForContext(t, ctx)
-	}
-	testCases := []func() error{
-		func() error {
-			parent_context, _ := context.WithTimeout(context.Background(), time.Duration(1.5*float64(timeUnit)))
-			_, err := Invoke(parent_context, nil, waitFunc, testOptions...)
-			return err
+var (
+	expected = &CallSettings{
+		time.Second * 1,
+		RetrySettings{
+			[]string{"hello", "world"},
+			BackoffSettings{
+				time.Second * 2,
```

Comment:
Doneish, since this is a test that checks the resolve behavior, and I removed this particular path, it's not useful to comment on names.